h264/parse.go

```diff
@@ -93,3 +93,69 @@ func readSe(r bitio.Reader) (int, error) {
 
 	return int(math.Pow(-1, float64(codeNum+1)) * math.Ceil(float64(codeNum)/2.0)), nil
 }
+
+// readMe parses a syntax element of me(v) descriptor, i.e. mapped
+// Exp-Golomb-coded element, using methods described in sections 9.1 and 9.1.2
+// in Rec. ITU-T H.264 (04/2017).
+func readMe(r bitio.Reader, chromaArrayType int, mpm macroblockPredictionMode) (uint, error) {
+	// Indexes to codedBlockPattern map.
+	var i1, i2, i3 int
+
+	// ChromaArrayType selects first index.
+	switch chromaArrayType {
+	case 1, 2:
+		i1 = 0
+	case 0, 3:
+		i1 = 1
+	default:
+		return 0, errors.New("invalid chromaArrayType")
+	}
+
+	// CodeNum from readUe selects second index.
+	i2, err := readUe(r)
+	if err != nil {
+		return 0, errors.Wrap(err, "error from readUe")
+	}
+
+	// Macroblock prediction mode selects third index.
+	switch mpm {
+	case intra4x4, intra8x8:
+		i3 = 0
+	case inter:
+		i3 = 1
+	default:
+		return 0, errors.New("invalid macroblock prediction mode")
+	}
+
+	return codedBlockPattern[i1][i2][i3], nil
+}
+
+// TODO: this should probably go in a separate consts file like consts.go or something
```

Comment:
Yes, please refactor into a separate file, or at the very least move to the top of the file.